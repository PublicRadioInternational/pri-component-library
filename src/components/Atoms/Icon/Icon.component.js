/**
 * @file Icon.component.js
 * Exports an icon component.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Icon.css';

/**
 * Component that renders a link, or a button with a click handler.
 */
const Icon = props => {
  const { svg, className, inline } = props;
  const icon = require(`./svg/${svg}.svg`); // eslint-disable-line
  return (
    <svg
      viewBox={icon.default.viewBox}
      className={inline ? styles.inlineSvg : className}
      fill="currentcolor"
    >
      <use xlinkHref={`#${icon.default.id}`} />
    </svg>
  );
};

Icon.propTypes = {
  // Worth automatically creating these from the files in ./svg sometime?
  svg: PropTypes.oneOf(['heart', 'envelope', 'search', 'volume', 'pri', 'play'])
    .isRequired,
  className: PropTypes.string,
  inline: PropTypes.bool
};

Icon.defaultProps = {
  className: null,
  inline: false
};

export default Icon;
