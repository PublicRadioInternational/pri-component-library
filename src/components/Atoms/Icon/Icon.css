/* import breakpoints */
@value header as bp-header from "../../00_global/breakpoints.css";

.inlineSvg {
  display: inline-block;
  margin-right: 8px;
  position: relative;
  top: 2px;
  width: 24px;
}

@media bp-header {
  .inlineSvg {
    width: 16px;
  }
}
