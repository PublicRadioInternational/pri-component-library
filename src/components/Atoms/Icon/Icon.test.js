/**
 * @file Button.test.js
 * Contains tests for Button.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import Icon from './Icon.component';

describe('<Icon />', () => {
  it('Matches the Icon snapshot', () => {
    const component = renderer.create(<Icon svg="heart" />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
