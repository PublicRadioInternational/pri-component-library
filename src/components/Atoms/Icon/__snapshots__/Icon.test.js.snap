// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> Matches the Icon snapshot 1`] = `
<svg
  aria-hidden={null}
  className={null}
  fill="currentcolor"
  viewBox={undefined}
>
  <use
    xlinkHref="#undefined"
  />
</svg>
`;
